src/main/scala/com/typesafe/sbt/osgi/Osgi.scala

```diff
@@ -44,62 +44,153 @@ private object Osgi {
+    headers: OsgiManifestHeaders,
+    additionalHeaders: Map[String, String],
+    fullClasspath: Seq[File],
+    artifactPath: File,
+    resourceDirectories: Seq[File],
+    embeddedJars: Seq[File],
+    explodedJars: Seq[File],
+    failOnUndecidedPackage: Boolean,
+    sourceDirectories: Seq[File],
+    packageOptions: scala.Seq[sbt.PackageOption],
+    useJVMJar: Boolean,
+    cacheStrategy: Option[CacheStrategy],
+    streams: TaskStreams): File =
+    withCache(headers,
+      additionalHeaders,
+      fullClasspath,
+      artifactPath,
+      resourceDirectories,
+      embeddedJars,
+      explodedJars,
+      failOnUndecidedPackage,
+      sourceDirectories,
+      packageOptions,
+      useJVMJar,
+      cacheStrategy) {
+        val builder = new Builder
+
+        if (failOnUndecidedPackage) {
+          streams.log.info("Validating all packages are set private or exported for OSGi explicitly...")
+          val internal = headers.privatePackage
+          val exported = headers.exportPackage
+          validateAllPackagesDecidedAbout(internal, exported, sourceDirectories)
+        }
+
+        builder.setClasspath(fullClasspath.toArray)
+
+        val props = headersToProperties(headers, additionalHeaders)
+        addPackageOptions(props, packageOptions)
+        builder.setProperties(props)
+
+        includeResourceProperty(resourceDirectories.filter(_.exists), embeddedJars, explodedJars) foreach (dirs =>
+          builder.setProperty(INCLUDERESOURCE, dirs))
+        bundleClasspathProperty(embeddedJars) foreach (jars =>
+          builder.setProperty(BUNDLE_CLASSPATH, jars))
+        // Write to a temporary file to prevent trying to simultaneously read from and write to the
+        // same jar file in exportJars mode (which causes a NullPointerException).
+        val tmpArtifactPath = file(artifactPath.absolutePath + ".tmp")
+        // builder.build is not thread-safe because it uses a static SimpleDateFormat.  This ensures
+        // that all calls to builder.build are serialized.
+        val jar = synchronized {
+          builder.build
+        }
+        val log = streams.log
+        builder.getWarnings.asScala.foreach(s => log.warn(s"bnd: $s"))
+        builder.getErrors.asScala.foreach(s => log.error(s"bnd: $s"))
+
+        if (!useJVMJar) jar.write(tmpArtifactPath)
+        else {
+          val tmpArtifactDirectoryPath = file(artifactPath.absolutePath + "_tmpdir")
+          IO.delete(tmpArtifactDirectoryPath)
+          tmpArtifactDirectoryPath.mkdirs()
+
+          val manifest = jar.getManifest
+          jar.writeFolder(tmpArtifactDirectoryPath)
+
+          def content = {
+            import _root_.java.nio.file._
+            import _root_.scala.collection.JavaConverters._
+            val path = tmpArtifactDirectoryPath.toPath
+            Files.walk(path).iterator.asScala.map(f => f.toFile -> path.relativize(f).toString).filterNot { case (_, p) => p == "META-INF/MANIFEST.MF" }.toTraversable
```

Comment:
@mdedetrich This `"META-INF/MANIFEST.MF" ` maybe cause an issue on Windows.

could you change to this like #115 does?
```scala
 Files.walk(path).iterator.asScala.map(f => f.toFile -> path.relativize(f))
              .collect { case (f, p) if p != (file("META-INF") / "MANIFEST.MF").toPath => (f, p.toString) }
```